libsplinter/src/service/scabbard/client/mod.rs

```diff
@@ -50,4 +53,92 @@ impl ScabbardClient {
             Ok(())
         }
     }
+
+    pub fn get_state_at_address(
+        &self,
+        circuit_id: &str,
+        service_id: &str,
+        address: &str,
+    ) -> Result<Option<Vec<u8>>, Error> {
+        let url = Url::parse(&format!(
+            "{}/{}/{}/{}/state/{}",
+            &self.url, SERVICE_TYPE, circuit_id, service_id, address
+        ))
+        .map_err(|err| Error::new_with_source("invalid URL", err.into()))?;
+
+        let request = Client::new().get(url);
+        let response = request
+            .send()
+            .map_err(|err| Error::new_with_source("request failed", err.into()))?;
+
+        if response.status().is_success() {
+            Ok(Some(response.json().map_err(|err| {
+                Error::new_with_source("failed to deserialize response body", err.into())
+            })?))
+        } else if response.status().as_u16() == 404 {
+            Ok(None)
+        } else {
+            let status = response.status();
+            let msg: ErrorResponse = response.json().map_err(|err| {
+                Error::new_with_source("failed to deserialize error response body", err.into())
+            })?;
+            Err(Error::new(&format!(
+                "failed to get state at address: {}: {}",
+                status, msg
+            )))
+        }
+    }
+
+    pub fn get_state_with_prefix(
+        &self,
+        circuit_id: &str,
+        service_id: &str,
+        prefix: Option<&str>,
```

Comment:
We should validate the prefix is an appropriate character set (hex) and set an upper bound on length (70) before constructing the URL. We could also verify here that it is an even number of characters.

Reply:
Added validation for the prefix here (valid hex, <70 characters) and added validation for the get_state_by_address method's address as well (just valid hex).